### PR TITLE
Added Login Screen ✅ 

### PR DESCRIPTION
**Pull request by Marco-Nagy**
Added Login Screen

+ create login ui screen
+ added login data and domain  layer
+ added login view model
+ added Error handler class

https://github.com/user-attachments/assets/94607afb-ce70-4bc0-a637-6299a46dcd42

### CODE REVIEW

**Review by SamyMohammad — APPROVED** [review]

File: lib/core/networking/error/error_handler.dart
Comment (mohamednabiltraining) [inline lib/core/networking/error/error_handler.dart:1]: please move these messages to presentation layer 
- don't return DIO exception inside Error Model
- instead create custom Exception s

File: lib/core/networking/error/error_contract.dart [hunk 1]
```diff
@@ -0,0 +1,22 @@
+
+import 'package:dio/dio.dart';
+
+import 'ErrorModel.dart';
+
+sealed class ErrorContract implements Exception{
+}
+
+class NetworkException implements ErrorContract{
+DioException exception;
```
Comment (mohamednabiltraining) [inline lib/core/networking/error/error_contract.dart:10]: depends on DIO package

File: lib/core/networking/error/error_handler.dart
Comment (mohamednabiltraining) [reply lib/core/networking/error/error_handler.dart:1]: i do this for 1, 2 3

File: lib/core/networking/error/error_contract.dart (hunk 1 above)
Comment (Marco-Nagy) [reply lib/core/networking/error/error_contract.dart:10]: is replacing Dio Exception with int status code good idea? 

`` class ErrorHandler {
  final String errorMessage;

  ErrorHandler(this.errorMessage);

  // Handle the exception and return an ErrorModel
  static ErrorModel handle(Exception error) {
    if (error is DioException) {
      return ErrorHandler.networkError(error, error.response?.statusCode);
    } else if (error is IOException) {
      return ErrorModel(message: "No internet connection. Please check your settings.");
    } else {
      return ErrorModel(message: "An unknown error occurred. Please try again.");
    }
  }

  // This method should return ErrorModel, not ErrorHandler
  static ErrorModel serverError(int? statusCode, ErrorModel response) {
    switch (statusCode) {
      case 401:
        return ErrorModel(message: "Unauthorized access. Please log in.");
      case 403:
        return ErrorModel(message: "Forbidden. You do not have permission to access this resource.");
      case 404:
        return ErrorModel(message: "Resource not found. Please check the URL and try again.");
      case 500:
        return ErrorModel(message: "Internal server error. Please try again later.");
      default:
        return ErrorModel(message: "An unexpected error occurred. Please try again.");
    }
  }

  static ErrorModel networkError(DioException error, int? statusCode) {
    switch (statusCode) {
      case 400:
        return ErrorModel(message: "Bad request. Please verify your input and try again.");
      case 401:
        return ErrorModel(message: "Unauthorized request. Please log in to continue.");
      case 403:
        return ErrorModel(message: "Access forbidden. You don't have permission for this action.");
      case 404:
        return ErrorModel(message: "Resource not found. The requested item does not exist.");
      case 408:
        return ErrorModel(message: "Connection timed out. Please check your internet connection.");
      case 500:
        return ErrorModel(message: "Internal server error. Please try again later.");
      case 502:
        return ErrorModel(message: "Bad Gateway. The server received an invalid response.");
      case 503:
        return ErrorModel(message: "Service Unavailable. The server is currently unable to handle the request.");
      case 504:
        return ErrorModel(message: "Gateway Timeout. The server took too long to respond.");
      default:
      // Ensure we convert response data to ErrorModel
        if (error.response?.data is Map<String, dynamic>) {
          final errorModel = ErrorModel.fromMap(error.response!.data as Map<String, dynamic>);
          return ErrorHandler.serverError(error.response?.statusCode, errorModel);
        } else {
          return ErrorModel(message: "An unexpected error occurred. Please try again.");
        }
    }
  }
}``

**Pull request merged**